hack/test-go.sh

```diff
@@ -159,11 +161,25 @@ else
     fi
 fi
 
+filtered_test_packages=
+excluded_test_packages=
+for package in ${test_packages}; do
+    if [[ ",${exclude_packages}," =~ ",${package}," ]]; then
```

Comment:
Not a fan of hack-ey set-contains checks like this, seen them fail too often. Can we set `$IFS` and iterate over the `$exclude_packages` list and do a direct comparison of elements?